### DISCUSSION TITLE
Wallet integration with external storage

### DISCUSSION THREAD

**Discussion by SethiShivam** [walt-id/.github#22] [Help] [answered]
Hi @dombrn @taminobaumann @philpotisk ,

I wanted to know if we can integrate wallet with external storage , like IPFS or a db ?
is there any reference available for that ?

Best Regards
Sethi Shivam

**Comment by SethiShivam** [comment]
REMINDER @taminobaumann @philpotisk

**Comment by taminobaumann** [comment, marked as answer]
Hey @SethiShivam,

yes, it is possible; however, we currently don't have a dedicated docs section for that feature yet. 

For the moment, I can point you to:

The implementation configuration is done in service-matrix.properties

For the keystore, see: id.walt.services.keystore.KeyStoreService=id.walt.services.keystore.SqlKeyStoreService
For the data store, see: id.walt.services.hkvstore.HKVStoreService=id.walt.services.hkvstore.FileSystemHKVStore:fsStore.conf
